osutil/disk.go

```diff
@@ -0,0 +1,61 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package osutil
+
+import (
+	"fmt"
+	"syscall"
+
+	"github.com/snapcore/snapd/strutil"
+)
+
+var syscalStatfs = syscall.Statfs
+
+type NotEnoughDiskSpaceError struct {
+	Path string
+	Delta int64
+}
+
+func (e *NotEnoughDiskSpaceError) Error() string {
+	return fmt.Sprintf("not enough free space in %s, requires %s", e.Path, strutil.SizeToStr(e.Delta))
```

Comment:
Maybe the error message should state that it requires _additional_ X bytes, to make it explicit that it's not considering the total size needed to store the file?

Reply:
Good point, done.